### DISCUSSION TITLE
[Snapcraft] versions need to be strings

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#765] [Q&A]
During my last release with JReleaser (v0.9.1), the Snapcraft package was initially rejected with

> Issues while validating snapcraft.yaml: The 'version' property does not match the required schema: snap versions need to be strings. They must also be wrapped in quotes when the value will be interpreted by the YAML parser as a non-string. Examples: '1', '1.2', '1.2.3', git (will be replaced by a git describe based version string).

I've [monkey-patched the YAML file](https://github.com/mthmulders/mcs-snap/commit/2fd6ea8ab1130b53e6de8a9faf38cde548ea7116) so the version number is enclosed in quotes, and then the Snapcraft package was built fine.

I've done a diff between JReleaser 0.9.1 and HEAD and it seems none of the Snapcraft related code has changed since - hence I suspect the issue will still be there. If so, would fixing this be considered for JReleaser 1.0.0? Maybe I can contribute?

**Comment by aalmiray** [comment]
Strange as JReleaser's snap has not been rejected. But yes, patches are welcome 😄

**Reply by mthmulders** [reply]
> Strange as JReleaser's snap has not been rejected.

Maybe because `1.0.0-M2` cannot be interpreted as a numeric value, while `0.2` can?

**Reply by aalmiray** [reply]
We've been using snap since `0.1.0`; didn't see the version number as flagged before.

**Reply by mthmulders** [reply]
I didn't have it with earlier releases (0.1.8, 0.1) either -- maybe they've updated their policies in the meantime?

Anyway, I'll propose a patch, just to be sure
